app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js

```diff
@@ -0,0 +1,256 @@
+  ${props =>
+    props.children.length &&
+    css`
+      border: ${th('borderWidth')} ${th('borderStyle')} ${th('colorBorder')};
+    `};
+`
+const FileBlock = styled(({ uploadError, ...rest }) => <Flex {...rest} />)`
+  border-top: ${th('borderWidth')} ${th('borderStyle')} ${th('colorBorder')};
+  ${props =>
+    props.uploadError
+      ? css`
+          margin: -1px -1px 0 -1px;
+          border: ${th('borderWidth')} ${th('borderStyle')} ${th('colorError')};
+        `
+      : css`
+          &:first-child {
+            border-top: none;
+          }
+        `};
+`
+const UploadSuccessIcon = styled(props => (
+  <Icon
+    iconName="CheckCircle"
+    overrideName="@pubsweet-pending.FileUpload.UploadSuccess"
+    {...props}
+  />
+))`
+  height: ${th('space.3')};
+  width: ${th('space.3')};
+  margin-right: ${th('space.2')};
+`
+
+const UploadFailureIcon = styled(props => (
+  <Icon
+    iconName="XCircle"
+    overrideName="@pubsweet-pending.FileUpload.UploadFailure"
+    {...props}
+  />
+))`
+  height: ${th('space.3')};
+  width: ${th('space.3')};
+  margin-right: ${th('space.2')};
+`
+const UploadControl = styled(Box).attrs({
+  width: 1 / 2,
+})`
+  display: none;
+  text-align: right;
+  &:first-child {
+    text-align: left;
+  }
+  ${media.tabletPortraitUp`
+    display: block;
+  `};
+`
+const Spinner = styled.span`
+  animation: full-rotation 1.1s infinite linear;
+  border: 2px solid ${th('colorPrimary')};
+  border-left: 2px solid ${th('colorBackground')};
+  border-radius: 50%;
+  display: inline-block;
+  height: ${th('space.3')};
+  overflow: hidden;
+  text-indent: -9999em;
+  transform: translateZ(0);
+  width: ${th('space.3')};
+  margin-right: ${th('space.2')};
+  @keyframes full-rotation {
+    0% {
+      transform: rotate(0deg);
+    }
+    100% {
+      transform: rotate(360deg);
+    }
+  }
+`
+
+function* fileUploadGenerator(files, uploadFile) {
+  for (let fileIndex = 0; fileIndex < files.length; ) {
+    yield {
+      upload: uploadFile(files[fileIndex].file),
+      fileId: files[fileIndex].id,
+    }
+    fileIndex += 1
+  }
+}
+
+class SupportingUpload extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      uploading: false,
+      files: props.files.map((file, index) => ({
+        file,
+        id: index,
+        success: true,
+      })),
+    }
+  }
+
+  synchronouslyUploadFiles = files => {
+    const { uploadFile } = this.props
+    const iterator = fileUploadGenerator(files, uploadFile)
+    const loop = result => {
+      if (result.done) {
+        this.setState({ uploading: false })
+      } else {
+        result.value.upload
+          .then(data => {
+            this.updateFileState(result.value.fileId, {
+              success: true,
+              loading: false,
+            })
+          })
+          .catch(() => {
+            this.updateFileState(result.value.fileId, {
+              error: true,
+              loading: false,
```

Comment:
What does this error state look like?

Reply:
red border and the same error icon as the manuscript upload failing